### PR TITLE
Add support for reading from HTTP broadcast streams

### LINKED ISSUE #53: GOTV+ Broadcasting
Hey,

same feature request as the old S1 parser https://github.com/saul/demofile/issues/373

The broadcast URLs for the current major are https://cstv.challengermode.com/major/primary and https://cstv.challengermode.com/major/secondary

I assume its almost identical to the CSGO one, convars are all the same at least

### PR DESCRIPTION
**Pull request by saul**
Closes #53

### CODE REVIEW

File: src/DemoFile/DemoParser.cs [hunk 1]
```diff
@@ -144,154 +127,15 @@ private void OnDemoPacket(CDemoPacket msg)
...
-
-    private async ValueTask<bool> MoveNextCoreAsync(EDemoCommands msgType, bool isCompressed, int size, CancellationToken cancellationToken)
+    internal bool ReadCommand(EDemoCommands msgType, bool isCompressed, ReadOnlySpan<byte> buffer, bool fireTimers)
     {
         IsReading = true;
 
         Debug.Assert(msgType is >= 0 and < EDemoCommands.DemMax, $"Unexpected demo command: {msgType}");
 
-        if (!IsSeeking)
+        //Console.WriteLine($"{msgType}({(int)msgType}) - {buffer.Length:N0} bytes");
```
Comment (saul) [inline src/DemoFile/DemoParser.cs:136]: - [ ] Remove debug statements

File: src/DemoFile/HttpBroadcastReader.cs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+public static class HttpBroadcastReader
+{
+    public static HttpBroadcastReader<TGameParser> Create<TGameParser>(DemoParser<TGameParser> demo, HttpClient httpClient)
+        where TGameParser : DemoParser<TGameParser>, new()
+    {
+        return new HttpBroadcastReader<TGameParser>(demo, httpClient);
+    }
+}
+
+public class HttpBroadcastReader<TGameParser>
```
Comment (saul) [inline src/DemoFile/HttpBroadcastReader.cs:20]: - [ ] Add tests for broadcasts

**Comment by saul** [comment]
@in0finite I'd be keen for you to try this out and see how it works for your use case.

One problem I had that I haven't fully resolved yet, is that DemoFile.Example.HttpBroadcast seems to churn through data faster than the data is received from the server. Sure it's a silly mistake somewhere but just a heads up.

**Comment by saul** [comment]
> @saul How do you test this now that there are no official CSTV+ servers ? I found this [project](https://github.com/FlowingSPDG/gotv-plus-go) which is "CSTV+ ingest" server, and also [this](https://www.reddit.com/r/GlobalOffensive/comments/qvk30n/run_your_own_csgo_server_with_gotv_like_it_was/) explanation from tournament organizer, but that's all.

I've been using Deadlock streams: main menu > Watch > pick any game, then read the http://dist1-ord1.steamcontent.com/tv/xxxxx in the console.

**Comment by in0finite** [comment]
I tested it using this approach:

### start CSTV+ ingest server:

`git clone https://github.com/FlowingSPDG/gotv-plus-go`
`cd gotv-plus-go`
`go run ./examples/inmemory/fiber/main.go -port 8080 -auth gopher`

### start CS server:

launch CS
type in Console: 
```
tv_enable 1
tv_broadcast_url "http://127.0.0.1:8080/gotv"
tv_broadcast_origin_auth "gopher"
tv_broadcast 1
```
Play => Practice => Competitive => (optionally enable Infinite warmup) => Start

### test parser:

you will see similar line in Console:
`[HLTV Broadcast] Broadcast[0] Recording started at http://127.0.0.1:8080/gotv/s58401682302641926t4945538495`

by starting the parser using that URL, following error appears:
```
System.NotSupportedException: 'Unrecognised serializer field: m_ProviderType'
    DemoFile.Sdk.CEntityInstance<TGameParser>.CreateFieldDecoder(DemoFile.SerializableField, DemoFile.Sdk.DecoderSet) in EngineSchema.cs
    DemoFile.Sdk.CBaseEntity.CreateFieldDecoder(DemoFile.SerializableField, DemoFile.Sdk.DecoderSet) in Schema.cs
    DemoFile.Sdk.CBaseModelEntity.CreateFieldDecoder(DemoFile.SerializableField, DemoFile.Sdk.DecoderSet) in Schema.cs
    DemoFile.Sdk.CBaseAnimGraph.CreateFieldDecoder(DemoFile.SerializableField, DemoFile.Sdk.DecoderSet) in Schema.cs
    DemoFile.Sdk.CBasePlayerWeapon.CreateFieldDecoder(DemoFile.SerializableField, DemoFile.Sdk.DecoderSet) in Schema.cs
    DemoFile.Sdk.DecoderSet.CreateDecoder.AnonymousMethod__1(DemoFile.SerializableField) in DecoderSet.cs
    [External Code]
    DemoFile.Sdk.DecoderSet.CreateDecoder<T>(DemoFile.SerializerKey) in DecoderSet.cs
    DemoFile.Sdk.DecoderSet.GetDecoder<T>(DemoFile.SerializerKey) in DecoderSet.cs
    DemoFile.Sdk.DeadlockDecoderSet.TryGetDecoderByName(string, out System.Type, out DemoFile.Sdk.SendNodeDecoder<object>) in Schema.cs
    ...
    [Call Stack Truncated]

```

**Pull request merged**